include/deal.II/base/index_set.h

```diff
@@ -691,7 +691,7 @@ class IndexSet
      * Dereferencing operator. The returned value is the index of the element
      * inside the IndexSet.
      */
-    size_type operator*() const;
+    const size_type &operator*() const;
```

Comment:
This is an unsigned int (or a 64 bit int). I don't think return by value is a problem. Is this just because the compiler doesn't know about this fact?

Reply:
Yes, `clang` can't do better than to observe that a copy is returned. Let me change all the places where we are binding to a reference instead.

Reply:
I pushed a new version.